tfx/components/example_gen/utils.py

```diff
@@ -346,12 +393,18 @@ def calculate_splits_fingerprint_span_and_version(
     logging.info('select span and version = (%s, %s)', select_span,
                  select_version)
     # Find most recent span and version for this split.
-    latest_span, latest_version = _retrieve_latest_span_version(
+    latest_span, latest_span_elems, latest_version, = _retrieve_latest_span_elems_version(
```

Comment:
my bad, i feel we should probably move split.pattern change in _retrieve_latest_span_elems_version, and just return span and version (to align, both span and version should be int, we might need to support {span:2} for the format (`01`, `02`), can be in future PR)

Reply:
should we just be setting them as int artifact properties?

Reply:
updated PR to move split.pattern update inside `_retrieve_latest_span_version()`

Reply:
also updated return type to be ints, and driver to set span and version as int properties